src/CyclotomicFac.jl

```diff
@@ -0,0 +1,125 @@
+@doc raw"""
+    CyclotomicFac{T <: PolyRingElem}
+
+The type representing a factorization into cyclotomic polynomials.
+"""
+struct CyclotomicFac{T <: PolyRingElem}
+  unit::T
+  fac::Dict{T, Int}
+  cyclo_fac::Dict{Int, Int}
+  function CyclotomicFac(f::Fac{T}) where T <: PolyRingElem
+    fac = Dict{T, Int}()
+    cyclo_fac = Dict{Int, Int}()
+    for (fact, expo) in f
+      b, n = is_cyclotomic_polynomial_with_data(fact)
```

Comment:
This function was only added in https://github.com/thofma/Hecke.jl/pull/1779, so this PR should also set a compat entry for Hecke to 0.35.13 (which will also automatically restrict Oscar to at least 1.3)

Reply:
Indeed, @SoongNoonien please do that

Reply:
Ok, we are now restricted to Oscar 1.3 anyway.

Reply:
Still, Oscar 1.3.0 is compatible with Hecke 0.35.10, but you use a feature from Hecke 0.35.13. This needs a compat entry for Hecke. See e.g. https://github.com/oscar-system/Oscar.jl/pull/4592/files#diff-72ed386c2a0cd1d23c0968297e70023ed98c22490d146dd89fc91f48369bad4d for an example of how to add a compat entry for an indirect dependency

Reply:
Ok, I see. Thanks!